### DISCUSSION TITLE
Change the name of movies

### DISCUSSION THREAD

**Discussion by Stefan765** [trix7777/m2strm#30] [Ideas]
Hello, unfortunately I have a few films with a prefix [DE] in my m3u, which is why emby cannot find the associated metadata. Is there a solution? Best regards

**Comment by trix7777** [comment]
Hi Stefan, I think that is a bug that you should report to the Emby developers 😂

You can also use a script that replaces characters in your m3u8, but that must happen before my m2strm processes it, ie, you need to download the m3u8 with another command, like curl or something.

**Reply by Stefan765** [reply]
Thank you for your answer! I think that's a bit too much for my programming skills 😏. Otherwise a big praise for your program, runs really smoothly!!!

**Comment by djvanes010** [comment]
Hello, first of all thanks for the m2strm program!

If you can replace 4K with UHD, you can also make a line for [DE] right? And also other filenames/characters; iptv providers are making more use of file / folder names, with (in my eyes not usefull) characters. Where can I put those lines (in program.cs), to replace [DE] with a space or something - and for example [NF]?

**Comment by djvanes010** [comment]
Ok, added some lines under:
//remove and replace chars -- this is done because GetInvalidFileNameChars behaves differently depending on OS
//here we do it on all OS to make the output more alike no matter OS

_Example_
DisneyPlus files are named: **_D+ - filename.strm_**
did add this line:
**.Replace("D+ -", "").Trim()**

After that, DisneyPlus files are named normal: **_filename.strm_**

**Reply by Stefan765** [reply]
Hello, unfortunately it doesn't work for me. The movie is called "DE - Disney", so my entry looks like this: .Replace("DE -", "").Trim()
Entered in the program.cs in the folder m2strm-3.0.2.6!
Whats wrong?
Thank you

**Reply by djvanes010** [reply]
Could you copy | paste a **full filename**? So I can look.

(Also, the folder naming is no problem in Emby. Only the files inside those folders, must not have "DE", "NL", "NF" or whatever prefix it has... It is not a real bug in Emby, it is how it is programmed for the metadata search and so on)

**Reply by djvanes010** [reply]
Oh, and did you also **(re-)build the m2strm.exe** after you make changes to program.cs?!

**Reply by Stefan765** [reply]
No, I only changed the Program.cs. How can I do the rebuild?

**Reply by djvanes010** [reply]
I do it with Visual Studio (Microsoft)

**Reply by Stefan765** [reply]
Unfortunately, I have no idea how this works! Is it very complicated?

**Reply by djvanes010** [reply]
Complicated no, but hard to explain in simple words ;-)

Here, that it mine m2strm.exe; you can download en copy this over de existing one.

https://we.tl/t-psbMAD2yBh

**Reply by djvanes010** [reply]
Oh, I did "DE" also in it...so it will delete the text "DE" for filenames ;-)

**Reply by Stefan765** [reply]
Thank you!! I will try later😊

**Reply by djvanes010** [reply]
Ok, let me know!

**Reply by djvanes010** [reply]
Latest version with more deleted prefixes in filenames and folder:

https://we.tl/t-x431zep9AH

**Reply by Stefan765** [reply]
so, I imported your exe file and it worked right away! Many thanks for your help!!!

**Reply by Stefan765** [reply]
Hello again, I have now added new films, but unfortunately the DE - is in front of it again! With the existing ones it works as intended?
Do you have a solution for this?
Best regards

**Reply by djvanes010** [reply]
Added new films to Emby?

Can you give an example please; full path and name of folder.

**Comment by Stefan765** [comment]
If a movie is added to my m3u8 and I run m2strm, the newly added movies are again preceded by DE -!
Emby cannot assign the metadata with the superior DE -.

**Reply by djvanes010** [reply]
I know, Emby and metadata. I'm using Emby also and that was the reason to find a solution. The one I use now (m2strm), is working the first run, but also the second, and so on...

You are sure you did the right one, m2strm.exe??

**Reply by Stefan765** [reply]
Yes, I deleted the folder and reloaded the m3u. There are a few movies with the DE - again!

**Reply by Stefan765** [reply]
![IMG_7979](https://github.com/user-attachments/assets/9871f3ac-1741-4344-b8e6-174b98c72dc6)
